### ISSUE TITLE
[BUG] Harvester installer should set hostname before getting node IP via DHCP 

### ISSUE THREAD

**Issue by w13915984028** [harvester/harvester#1682]
**Describe the bug**

The user input "hostname" in harvester installation is not set before getting node IP via DHCP, it causes node IP change in KVM environment.

**To Reproduce**
1. Create one VM in KVM, use default "NAT" network
2. Install Harvester from ISO in this VM, set hostname, set DHCP mode getting node IP
3. After installation, node IP changed 

**Expected behavior**

node IP should keep same

**Support bundle**

**Notice, the KVM DHCP list, it cares about "Hostname",
The installation uses "rancher" to apply IP, after rebooting, the Harvester uses user-input hostname to apply IP,
even though MAC is not changed, but KVM DHCP server gives different IP**

![image](https://user-images.githubusercontent.com/31133476/146196233-fe4ce2d6-b8fe-4cd5-9c20-957239f70bd8.png)
![image](https://user-images.githubusercontent.com/31133476/146196291-7984f2e7-6708-450c-8633-20830ee68b1a.png)
![image](https://user-images.githubusercontent.com/31133476/146196333-24534146-fd0d-45af-8310-78821bd61552.png)

**Environment:**
 - Harvester ISO version:  V1.0.0 master-head
 - Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630): 
 Provo DAY0 HP server, single rack, HOST OS. SUSE SLE15 SP2

**Additional context**
refer:
https://github.com/harvester/harvester/issues/1668
reproduce :
https://github.com/w13915984028/harvester-develop-summary/issues/1

**Comment by w13915984028** [comment]
@guangbochen @johnliu55tw

**Comment by yasker** [comment]
@guangbochen can we move this issue to v1.0.1?

**Comment by w13915984028** [comment]
Even when keep the hostname as "rancher", it looks the NODE IP from KVM still changed, needs further investigation

**but the "Client ID or DUID" has a change**, we did not notice it before

```
ff:00:f7:72:bf:00:01:00:01:29:4e_:04:bd_:52:54:00:f7:72:bf

ff:00:f7:72:bf:00:01:00:01:29:4e:_06:ff_:52:54:00:f7:72:bf
```

```
virsh # net-dhcp-leases --network default
 Expiry Time           MAC address         Protocol   IP address           Hostname       Client ID or DUID
----------------------------------------------------------------------------------------------------------------------------------------------------
 2021-12-16 10:23:11   46:6c:14:89:e8:ee   ipv4       192.168.122.225/24   -              -

 2021-12-16 10:08:14   52:54:00:f7:72:bf   ipv4       192.168.122.71/24    rancher        ff:00:f7:72:bf:00:01:00:01:29:4e:04:bd:52:54:00:f7:72:bf

 2021-12-16 10:17:38   52:54:00:f7:72:bf   ipv4       192.168.122.72/24    rancher        ff:00:f7:72:bf:00:01:00:01:29:4e:06:ff:52:54:00:f7:72:bf
```

![image](https://user-images.githubusercontent.com/31133476/146391154-ab9a9d78-322f-4864-970b-43c504ba5341.png)

![image](https://user-images.githubusercontent.com/31133476/146391298-f9ff9d28-486d-4d08-baf8-f40df1316fb9.png)

**Comment by johnliu55tw** [comment]
`wicked` by default uses [DUID-LLT](https://datatracker.ietf.org/doc/html/rfc3315#section-9.2) as the client identifier to request IP from DHCP server, and the one generated during the installation time is different from the one generated on first-boot after installation because **the time** is involved in generating the identifier (LTT stands for **L**ink-**L**ayer address plus **T**ime).

One way to test this (and probably a solution) is to copy wicked runtime data (`/var/lib/wicked/*`) from installation-time to the installed system by adding these lines to `/usr/sbin/harv-install`, anywhere after `do_mount`:

```sh
wicked_state="${TARGET}/usr/local/.state/var-lib-wicked.bind"
mkdir -p ${wicked_state}
cp /var/lib/wicked/* ${wicked_state}
```

As far as I can tell, `/var/lib/wicked/duid.xml` is absolutely related to DUID creation. As for other files, I also copied them just in case.

The other solution could be changing the algorithm for generating [DUID](https://datatracker.ietf.org/doc/html/rfc3315#section-9), such as [DUID-EN](https://datatracker.ietf.org/doc/html/rfc3315#section-9.3) or [DUID-LL](https://datatracker.ietf.org/doc/html/rfc3315#section-9.4), or fallback to [rfc2131](https://www.mankier.com/5/wicked-config#DHCP4_Supplicant_Options-create-cid) which only uses MAC to create client identifier, or just disable client identifier entirely and let DHCP server uses MAC plus Hostname to recognize the machine.

See [wicked DHCP4 Supplicant Options/create-cid](https://www.mankier.com/5/wicked-config#DHCP4_Supplicant_Options-create-cid) for more information.

**Comment by TachunLin** [comment]
Verified **fixed** on master-ebde990d-head on `single node` harvester scenario.

### Result

1. Node IP and VIP remain unchanged before, during and after installation
2. No rancher hostname used during the installation

* Before installation, (set hostname)
  ![image](https://user-images.githubusercontent.com/29251855/149074931-e1020617-1d9d-49f5-9351-61bcb921e1b9.png)

  ![image](https://user-images.githubusercontent.com/29251855/149075048-5e306c28-7ee4-4d94-84a5-8e21e64218be.png)

  ![image](https://user-images.githubusercontent.com/29251855/149075281-2eda76a8-1310-46c6-9cd7-19a87699b8fa.png)

* After install before login harvester
  ![image](https://user-images.githubusercontent.com/29251855/149077038-61105e3e-ca53-4f2f-9229-4ec45f4d0ee5.png)
  
  ![image](https://user-images.githubusercontent.com/29251855/149076914-725c3351-7df6-40c1-a00e-d5154c55b92a.png)

* Login harvester after installaion

  ![image](https://user-images.githubusercontent.com/29251855/149077536-d8e47eac-285f-4b72-b840-f7d2fb36834d.png)

  ![image](https://user-images.githubusercontent.com/29251855/149078201-a2e6383a-4d15-47af-a3b6-c3fc4e20b3ad.png)

  ```
  3: harvester-mgmt: <BROADCAST,MULTICAST,MASTER,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
      link/ether 52:54:00:ee:fe:b6 brd ff:ff:ff:ff:ff:ff
      inet 192.168.122.141/24 brd 192.168.122.255 scope global harvester-mgmt
         valid_lft forever preferred_lft forever
      inet 192.168.122.69/32 scope global harvester-mgmt
         valid_lft forever preferred_lft forever
  ```

### Test Information
* Test Environment: 1 node1 harvester on local kvm
* Harvester version: master-ebde990d-head (22/01/12)

### Verify Steps

1. ISO install single node harvester
2. Set DHCP to get node IP
3. Input hostname
4. Set DHCP to get VIP 
5. Run virsh -> `net-dhcp-leases --network default` to check DHCP leased ip
3. Check DHCP lease during installation
4. After installation check the

**Comment by TachunLin** [comment]
Verified **fixed** on master-ebde990d-head on `3 nodes` harvester scenario.

### Result

1. Node IP and VIP remain unchanged before, during and after installation
2. No rancher hostname used during the installation

* **1st node**  -> Before installation, (set host name)
 ![image](https://user-images.githubusercontent.com/29251855/149082826-b6d7cd4e-60d9-4b76-9c0e-fc71ea03973b.png)
 ![image](https://user-images.githubusercontent.com/29251855/149082929-77f7188b-a5b4-4344-ac69-f91acf960de3.png)
 ![image](https://user-images.githubusercontent.com/29251855/149083121-f98e6f8b-1bd0-4aeb-bfe1-24e04369a2d5.png)

* **1st node**  -> After installation 
  ![image](https://user-images.githubusercontent.com/29251855/149098183-2dbebafd-4c9a-404d-8952-56feae02deb6.png)
  

* **2nd node**  -> Before installation, (set host name)
  ![image](https://user-images.githubusercontent.com/29251855/149099311-15fabfca-b05d-4f0a-8867-0bc9abbe9292.png)
  ![image](https://user-images.githubusercontent.com/29251855/149099720-adcd4607-4db4-447c-952c-1b6e2f007271.png)
  
* **2nd node**  -> After installation 
  ![image](https://user-images.githubusercontent.com/29251855/149103263-6581cd9d-3a0a-4970-9f93-ab62350f784b.png)
  ![image](https://user-images.githubusercontent.com/29251855/149103508-1e26588a-2b2a-494e-a4a9-dc3f56214b24.png)
  ![image](https://user-images.githubusercontent.com/29251855/149104043-946cb2a8-54c3-4424-88ea-5c486476def3.png)

* **3rd node**  -> Before installation, (set host name)
  ![image](https://user-images.githubusercontent.com/29251855/149106090-6ad813b8-808a-43ed-9c62-995254581213.png)
  ![image](https://user-images.githubusercontent.com/29251855/149106368-45650e99-4329-49f4-8772-0bf227c98be4.png)

* **3rd node**  -> After installation 
  ![image](https://user-images.githubusercontent.com/29251855/149110067-19a6ea50-1808-448d-bc47-a5a5674a373d.png)
  ![image](https://user-images.githubusercontent.com/29251855/149111654-0240f4cf-69e6-44a8-9abb-c7e2b5373950.png)
  ![image](https://user-images.githubusercontent.com/29251855/149112628-6ef0f65c-adc5-4a27-ba9a-056d7b65bc41.png)

**Comment by TachunLin** [comment]
Verified fixed on master-ebde990d-head both single and 3 nodes harvester scenario. Close this issue
